File: implementations/wrap-assemblyscript/src/__tests__/cases/001-module-functions/output/Module/wrapped.ts
import { wrap_load_env } from "@polywrap/wasm-as";
import {
  deserializefunction1Args,
  serializefunction1Result,
  deserializefunction2Args,
  serializefunction2Result
} from "./serialization";
import { ModuleBase } from "./module";
import * as Types from "..";

export function function1Wrapped(module: ModuleBase, argsBuf: ArrayBuffer, env_size: u32): ArrayBuffer {
  const args = deserializefunction1Args(argsBuf);

  const result = module.function1(
    {
      arg1: args.arg1,
      arg2: args.arg2
    }
  );
  return serializefunction1Result(result);
}

export function function2Wrapped(module: ModuleBase, argsBuf: ArrayBuffer, env_size: u32): ArrayBuffer {
  const args = deserializefunction2Args(argsBuf);

  const result = module.function2(
    {
      arg1: args.arg1,
      arg2: args.arg2
    }
  );
  return serializefunction2Result(result);
}

File: implementations/wrap-rust/src/__tests__/cases/001-module-functions/output/module/wrapped.rs
use serde::{Deserialize, Serialize};
use polywrap_wasm_rs::{
    BigInt,
    BigNumber,
    Map,
    JSON,
    wrap_load_env,
    to_vec,
    from_slice,
    JSONString,
    BigIntWrapper,
    ByteBuf
};
use crate::{ModuleTrait, Module};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ArgsFunction1 {
    pub arg1: u32,
    pub arg2: bool,
}

pub fn function1_wrapped(args: &[u8], env_size: u32) -> Vec<u8> {
    match from_slice::<ArgsFunction1>(args) {
        Ok(args) => {
            let result = Module::function1(ArgsFunction1 {
                arg1: args.arg1,
                arg2: args.arg2,
            });
            match result {
                Ok(res) => {
                    to_vec(&res).unwrap()
                }
                Err(e) => {
                    panic!("{}", e.to_string())
                }
            }
        }
        Err(e) => {
            panic!("{}", e.to_string())
        }
    }
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ArgsFunction2 {
    pub arg1: Option<i32>,
    pub arg2: Option<ByteBuf>,
}

pub fn function2_wrapped(args: &[u8], env_size: u32) -> Vec<u8> {
    match from_slice::<ArgsFunction2>(args) {
        Ok(args) => {
            let result = Module::function2(ArgsFunction2 {
                arg1: args.arg1,
                arg2: args.arg2,
            });
            match result {
                Ok(res) => {
                    to_vec(&res).unwrap()
                }
                Err(e) => {
                    panic!("{}", e.to_string())
                }
            }
        }
        Err(e) => {
            panic!("{}", e.to_string())
        }
    }
}
